src/anomalib/models/components/base/anomaly_module.py

```diff
@@ -213,6 +192,42 @@ def default_post_processor(self) -> PostProcessor:
               Please override the default_post_processor method in the model implementation."
         raise NotImplementedError(msg)
 
+    @staticmethod
+    def default_evaluator() -> Evaluator:
+        """Default evaluator.
+
+        Override in subclass for model-specific evaluator behaviour.
+        """
+        image_auroc = AUROC(fields=["pred_score", "gt_label"], prefix="image_")
+        image_f1score = F1Score(fields=["pred_label", "gt_label"], prefix="image_")
+        pixel_auroc = AUROC(fields=["anomaly_map", "gt_mask"], prefix="pixel_")
+        pixel_f1score = F1Score(fields=["pred_mask", "gt_mask"], prefix="pixel_")
+        test_metrics = [image_auroc, image_f1score, pixel_auroc, pixel_f1score]
+        return Evaluator(test_metrics=test_metrics)
```

Comment:
In many cases it is not necessary to compute validation metrics. The main purpose of validation in anomaly models is to compute the post-processing metrics such as adaptive threshold and normalization stats, and early stopping in multi-epoch models. The post-processing metrics are covered by the `PostProcessor` class, so the most common reason to compute metrics over the validation set is early stopping. But not all models in Anomalib use or can use early stopping. For example, we have many feature extraction based models which only run for 1 epoch. Computing validation metrics for these models would not be very useful, while it may significantly slow down the training process, as pixel metrics may require some heavy computation.

I don't think the early stopping requirement of some models warrants the unnecessary computation of validation metrics for all other models. If a model uses early stopping by default, then the model implementer should just make sure that the early stopping metric is included by default. If a user wants to enable early stopping for some model, they should add the early stopping metric to the evaluator.

Reply:
Okay, makes sense. Early stopping is usually not used by original authors, and I agree that this can be handled by the user.